src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -400,14 +494,21 @@ protected virtual TypeSymbolWithAnnotations SubstituteType(AbstractTypeMap typeM
             {
                 return newTypeWithModifiers;
             }
-            return new NonLazyType(
+            return TypeSymbolWithAnnotations.CreateNonLazyType(
                 newTypeWithModifiers.TypeSymbol,
                 newTypeWithModifiers.NonNullTypesContext,
                 isAnnotated: newIsAnnotated,
                 newCustomModifiers.Concat(newTypeWithModifiers.CustomModifiers));
         }
 
-        public virtual void ReportDiagnosticsIfObsolete(Binder binder, SyntaxNode syntax, DiagnosticBag diagnostics)
+        public void ReportDiagnosticsIfObsolete(Binder binder, SyntaxNode syntax, DiagnosticBag diagnostics) => _extensions.ReportDiagnosticsIfObsolete(this, binder, syntax, diagnostics);
```

Comment:
>=> [](start = 109, length = 2)

nit: I don't think we've established a style, but I'd suggest breaking the lines with `=>`.
```C#
public void ReportDiagnosticsIfObsolete(Binder binder, SyntaxNode syntax, DiagnosticBag diagnostics) 
    => _extensions.ReportDiagnosticsIfObsolete(this, binder, syntax, diagnostics);
```